Common/TableProducer/PID/pidTPC.cxx

```diff
@@ -105,53 +107,39 @@ struct tpcPid {
-
-      path = ccdbPath.value + "/" + sigmaname.value;
-      LOG(info) << "Loading exp. sigma parametrization from CCDB, using path: " << path << " for timestamp " << timestamp.value;
-      response.LoadParam(DetectorResponse::kSigma, ccdb->getForTimeStamp<Parametrization>(path, timestamp.value));
+    const TString fname = paramfile.value;
+    if (fname != "") { // Loading the parametrization from file
+      LOGP(info, "Loading TPC response from file {}", fname);
+      try {
+        std::unique_ptr<TFile> f(TFile::Open(fname, "READ"));
+        f->GetObject("Response", response);
+        response->SetUseDefaultResolutionParam(false);
+      } catch (...) {
+        LOGP(info, "Loading the TPC PID Response from file {} failed!", fname);
+      };
+    } else {
+      const std::string path = ccdbPath.value;
+      const auto time = timestamp.value;
+      ccdb->setURL(url.value);
+      ccdb->setTimestamp(time);
+      ccdb->setCaching(true);
+      ccdb->setLocalObjectValidityChecking();
+      ccdb->setCreatedNotAfter(std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count());
+      response = ccdb->getForTimeStamp<o2::pid::tpc::Response>(path, time);
+      LOGP(info, "Loading TPC response from CCDB, using path: {} for timestamp {}", path, time);
     }
   }
 
-  template <o2::track::PID::ID pid>
-  using ResponseImplementation = o2::pid::tpc::ELoss<Trks::iterator, pid>;
   void process(Coll const& collisions, Trks const& tracks)
   {
-    constexpr auto responseEl = ResponseImplementation<PID::Electron>();
-    constexpr auto responseMu = ResponseImplementation<PID::Muon>();
-    constexpr auto responsePi = ResponseImplementation<PID::Pion>();
-    constexpr auto responseKa = ResponseImplementation<PID::Kaon>();
-    constexpr auto responsePr = ResponseImplementation<PID::Proton>();
-    constexpr auto responseDe = ResponseImplementation<PID::Deuteron>();
-    constexpr auto responseTr = ResponseImplementation<PID::Triton>();
-    constexpr auto responseHe = ResponseImplementation<PID::Helium3>();
-    constexpr auto responseAl = ResponseImplementation<PID::Alpha>();
-
     // Check and fill enabled tables
-    auto makeTable = [&tracks](const Configurable<int>& flag, auto& table, const DetectorResponse& response, const auto& responsePID) {
+    auto makeTable = [&tracks, &collisions](const Configurable<int>& flag, auto& table, const o2::pid::tpc::Response& response, const o2::track::PID::ID pid) {
       if (flag.value == 1) {
         // Prepare memory for enabled tables
         table.reserve(tracks.size());
         for (auto const& trk : tracks) { // Loop on Tracks
-          const float separation = responsePID.GetSeparation(response, trk);
+          auto collision = collisions.iteratorAt(trk.collisionId());
+          const float separation = response.GetExpectedSigma(collision, trk, pid);
```

Comment:
Is this the Nsigma or the expected resolution?

Reply:
This is the expected resolution. I changed it to the number of sigma. Thanks for spotting this.

Reply:
Thanks to you!